packages/typed-ipc/src/main.ts

```diff
@@ -8,42 +8,62 @@ export * from './types'
  * Typed listener for Electron `ipcMain`.
  */
 export class IpcListener<T extends IpcEventMap> {
-  private listeners: string[] = []
-  private handlers: string[] = []
+  private listeners: Map<string, Set<(...args: any[]) => void>> = new Map()
+  private handlers: Set<string> = new Set()
```

Comment:
Using a `Set` fixes a subtle issue. With an array, it's possible that we could end up with duplicates channels in the array.

You can only have 1 handler per channel, but the `handle` method will add the channel to the array even if a handler already exists for the channel.

When we `dispose`, we may end up attempting to remove listeners for a channel multiple times.

I don't think this causes any problems in practice, but using `Set` prevents the situation from happening in the first place.